### PR TITLE
chore(composer): Manage browser dependencies via composer

### PR DESCRIPTION
**Pull request by ewinslow**
See https://github.com/francoispluchino/composer-asset-plugin for details.

### CODE REVIEW

**Comment by ewinslow** [comment]
This installs fine on my environment. Don't know why it isn't working for Travis... giving up for now.

**Comment by ewinslow** [comment]
Made some tweaks. Don't understand why it works this way now, but it does...

**Comment by ewinslow** [comment]
OK, so it seems like the determining factor is whether we try to install it in a pre-install-cmd script.

Travis didn't seem to like that, so will just install with a separate "setup" command or something.

**Comment by ewinslow** [comment]
Or just do it manually, since that's how everyone else is doing to have to install Elgg for now until we find a better solution...

**Comment by ewinslow** [comment]
@Elgg/owners I find it surprising I haven't gotten any comments on this yet.

WDYT?

File: composer.json [hunk 1]
```diff
@@ -19,13 +19,75 @@
...
+            "type": "package",
+            "package": {
+                "name": "bower-asset/jquery-jeditable",
+                "version": "1.7.3",
+                "dist": {
+                    "type": "zip",
+                    "url": "https://github.com/tuupola/jquery_jeditable/archive/1.7.3.zip"
+                }
+            }
+        }
```
Comment (ewinslow) [inline composer.json]: These packages I had to hardcode in are quite old. We should ditch/upgrade them when we can...

**Comment by mrclay** [comment]
Seems like a great move

**Comment by ewinslow** [comment]
PSA: This will require you to do a:

```
composer global require "fxp/composer-assets-plugin:~1.0"
```

in order to install succesfully from now on (which means the release scripts are also broken until that command is run).

**Comment by mrclay** [comment]
How mature/supported do we judge the bower-assets plugin to be? It makes me a little nervous. If you upgrade a resource does `composer update` really handle it all? Do we need to version the URLs or anything?

**Comment by ewinslow** [comment]
1 year old. 30,000 download per month. Just got to 1.0. Ran into several problems, all of which were addressed by docs. Main thing was just needing to install globally.

By my accounting, this is reliable software.

`composer update` really handles it all, so far as I can tell.

We feed the assets through simple cache so that takes care of versioning. Simplecache may cache bust too aggressively, but that's something that we should just take care of on our end hopefully transparently for plugin devs...

**Comment by mrclay** [comment]
Thanks.

**Comment by ewinslow** [comment]
No problem. Going to pull this in Monday if I don't hear any objections

**Comment by ewinslow** [comment]
Excited about this + views rewrite making it easy to pull in more third party front end code...

**Comment by juho-jaakkola** [comment]
LGTM, but there seem to be conflicts

**Comment by ewinslow** [comment]
Got a couple LGTMs and I believe I addressed all objections. Pulling the trigger on this one...

**Comment by mrclay** [comment]
I get this:
```
Could not fetch https://api.github.com/repos/requirejs/text/contents/bower.json?ref=712f2d862693107cf993adb0c01178a9fce714c8, please create a GitHub OAuth token to go over the API rate limit
Head to https://github.com/settings/tokens/new?scopes=repo&description=Composer+on+steves-mbp+2015-06-05+1245
to retrieve a token. It will be stored in "/Users/steve/.composer/auth.json" for future use by Composer.
```

**Comment by mrclay** [comment]
Got the token, installation continuing...

**Comment by ewinslow** [comment]
All good?

**Pull request merged**